src/wallet/wallet.cpp

```diff
@@ -4133,6 +4133,8 @@ bool CWallet::UpgradeWallet(int version, bilingual_str& error)
             return false;
         }
     }
+
+    version = GetVersion(); // return actual updated version
```

Comment:
in commit 2310fe9685246b0d505495ee98d38c82a1730293: would be a smaller diff and less complexity if this was moved to the caller, no?

Reply:
Yes, I tried removing the first commit and just calling GetVersion() in `RPCHelpMan upgradewallet()` but the first and last tests failed. Maybe a lock order/data race issue.

AssertionError: not({'wallet_name': '', 'previous_version': 159900, 'current_version': 169900, 'result': 'Wallet upgraded successfully from version 159900 to version 0.'} == {'wallet_name': '', 'previous_version': 159900, 'current_version': 169900, 'result': 'Wallet upgraded successfully from version 159900 to version 169900.'})

AssertionError: not({'wallet_name': '', 'previous_version': 139900, 'current_version': 169900, 'result': 'Wallet upgraded successfully from version 139900 to version 159900.'} == {'wallet_name': '', 'previous_version': 139900, 'current_version': 159900, 'result': 'Wallet upgraded successfully from version 139900 to version 159900.'})

Reply:
Hm, will try again.

Reply:
I didn't update one of the version variables the first time. Seems good.